core/src/stages/connect.cpp

```diff
@@ -161,13 +164,17 @@ void Connect::compute(const InterfaceState& from, const InterfaceState& to) {
 
 		robot_trajectory::RobotTrajectoryPtr trajectory;
 		success = pair.second->plan(start, end, jmg, timeout, trajectory, path_constraints);
+		sub_trajectories.push_back(trajectory);
 
-		// Do not push partial solutions
+		// Check if the end goal and the last waypoint are close. The trajectory should be marked as failure otherwise.
 		if (success) {
-			sub_trajectories.push_back(trajectory);
-		} else {
-			// Pushing a nullptr instead of a failed trajectory.
-			sub_trajectories.push_back(nullptr);
+			const auto distance = end->getCurrentState().distance(trajectory->getLastWayPoint());
+			if (distance > 0.05) {
```

Comment:
Please make this be a parameter of the stage, so it can be tuned at stage creation time?